File: swift/PassengerApp/PassengerApp/GeneralFiles/SetUserData.swift
//
//  SetUserData.swift
//  PassengerApp
//
//  Created by NEW MAC on 11/05/17.
//  Copyright © 2017 V3Cube. All rights reserved.
//

import UIKit

class SetUserData: NSObject {

    var viewControlller:UIViewController!
    var userProfileJson:NSDictionary!
    var isStoreUserId = true
    let generalFunc = GeneralFunctions()
    
    init(uv: UIViewController, userProfileJson:NSDictionary, isStoreUserId:Bool) {
        self.viewControlller = uv
        self.userProfileJson = userProfileJson
        self.isStoreUserId = isStoreUserId
        super.init()
        
        setData()
    }
    
    
    
    func setData(){
        let isLanguageCodeChanged = self.userProfileJson!.get("changeLangCode")
        
        if(isLanguageCodeChanged == "Yes"){
            GeneralFunctions.saveValue(key: Utils.languageLabelsKey, value: self.userProfileJson!.getObj("UpdatedLanguageLabels"))
            GeneralFunctions.saveValue(key: Utils.LANGUAGE_CODE_KEY, value: self.userProfileJson!.get("vLanguageCode") as AnyObject)
            GeneralFunctions.saveValue(key: Utils.LANGUAGE_IS_RTL_KEY, value: self.userProfileJson!.get("langType") as AnyObject)
            GeneralFunctions.saveValue(key: Utils.GOOGLE_MAP_LANGUAGE_CODE_KEY, value: self.userProfileJson!.get("vGMapLangCode") as AnyObject)
            
            if(self.userProfileJson!.isKeyExist("LIST_LANGUAGES")){
                GeneralFunctions.saveValue(key: Utils.LANGUAGE_LIST_KEY, value: self.userProfileJson!.getArrObj("LIST_LANGUAGES"))
            }
            
            if(self.userProfileJson!.isKeyExist("LIST_CURRENCY")){
                GeneralFunctions.saveValue(key: Utils.CURRENCY_LIST_KEY, value: self.userProfileJson!.getArrObj("LIST_CURRENCY"))
            }
        }
        
        if(isStoreUserId){
            GeneralFunctions.saveValue(key: Utils.iMemberId_KEY, value: self.userProfileJson!.getObj(Utils.message_str).get("iUserId") as AnyObject)
            GeneralFunctions.saveValue(key: Utils.isUserLogIn, value: "1" as AnyObject)
        }
    }
}

File: android/PassengerApp/app/src/main/java/com/general/files/SetUserData.java
package com.general.files;

import android.content.Context;

import com.splunk.mint.Mint;
import com.utils.CommonUtilities;
import com.utils.Utils;

/**
 * Created by Admin on 29-06-2016.
 */
public class SetUserData {
    String userProfileJson;

    GeneralFunctions generalFunc;
    boolean isStoreUserId = true;
    Context mContext;

    public SetUserData(String userProfileJson, GeneralFunctions generalFunc, Context mContext, boolean isStoreUserId) {
        this.userProfileJson = userProfileJson;
        this.generalFunc = generalFunc;
        this.mContext = mContext;
        this.isStoreUserId = isStoreUserId;
        setData();

    }

    public SetUserData(String userProfileJson, GeneralFunctions generalFunc, boolean isStoreUserId) {
        this.userProfileJson = userProfileJson;
        this.generalFunc = generalFunc;
        this.isStoreUserId = isStoreUserId;

        setData();
    }

    public void setData() {
        String isLanguageCodeChanged = generalFunc.getJsonValue("changeLangCode", userProfileJson);
        if (isLanguageCodeChanged.equals("Yes")) {

            generalFunc.storedata(CommonUtilities.languageLabelsKey, generalFunc.getJsonValue("UpdatedLanguageLabels", userProfileJson));
            generalFunc.storedata(CommonUtilities.LANGUAGE_CODE_KEY, generalFunc.getJsonValue("vLanguageCode", userProfileJson));
            generalFunc.storedata(CommonUtilities.LANGUAGE_IS_RTL_KEY, generalFunc.getJsonValue("langType", userProfileJson));

            generalFunc.storedata(CommonUtilities.GOOGLE_MAP_LANGUAGE_CODE_KEY, generalFunc.getJsonValue("vGMapLangCode", userProfileJson));

            Utils.setAppLocal(mContext);

            if (!generalFunc.getJsonValue("LIST_LANGUAGES", userProfileJson).equals("")) {
                generalFunc.storedata(CommonUtilities.LANGUAGE_LIST_KEY, generalFunc.getJsonValue("LIST_LANGUAGES", userProfileJson));
            }

            if (!generalFunc.getJsonValue("LIST_CURRENCY", userProfileJson).equals("")) {
                generalFunc.storedata(CommonUtilities.CURRENCY_LIST_KEY, generalFunc.getJsonValue("LIST_CURRENCY", userProfileJson));
            }

        }

        if (isStoreUserId == true) {
            String memberId = generalFunc.getJsonValue("iUserId", generalFunc.getJsonValue(CommonUtilities.message_str, userProfileJson));
            generalFunc.storeUserData(memberId);

            Mint.addExtraData("iMemberId", "" + generalFunc.getMemberId());
        }

    }
}
